lib/services/crawler/crawler.go

```diff
@@ -0,0 +1,153 @@
+package crawler
+
+import (
+	"database/sql"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"gorm.io/gorm"
+	"net/url"
+	"strings"
+
+	"github.com/markgravity/golang-ic/helpers/log"
+	"github.com/markgravity/golang-ic/lib/models"
+
+	"github.com/gocolly/colly/v2"
+	"github.com/gocolly/colly/v2/extensions"
+)
+
+type Crawler struct {
+	DB        *gorm.DB
+	Keyword   *models.Keyword
+	Collector *colly.Collector
+
+	parsingResult *ParsingResult
+}
+
+type ParsingResult struct {
+	HTMLCode        string
+	NonAdwordLinks  []string
+	AdwordLinks     []string
+	ShopAdwordLinks []string
+}
+
+var selectors = map[string][]string{
+	"nonAds":      {"#search .yuRUbf > a", ".ZINbbc.xpd .kCrYT > a"},
+	"ads":         {"#tads .d5oMvf > a", ".uEierd .ZINbbc a.C8nzq.BmP5tf"},
+	"mobileLinks": {".ezO2md a.fuLhoc.ZWRArf"},
+	"mobileAds":   {"span.dloBPe"},
```

Comment:
In case the current ordering has no perticular purpose, should we order by alphabetical ordering?

Reply:
I prefer to keep the order from Google as they are ordering by their algorithm, right 🤔